### DISCUSSION TITLE
Error with account type in aptos.transaction.sign

### DISCUSSION THREAD

**Discussion by MartinYeung5** [aptos-labs/aptos-developer-discussions#369] [Dev Doc and Tutorial Feedback]
Hi team, i tried to use aptos.transaction.sign, 

The following is my code:
import { useWallet } from "@aptos-labs/wallet-adapter-react";
const { account, signAndSubmitTransaction } = useWallet();
 const owner1 = account;

const bobSenderAuthenticator = aptos.transaction.sign({
    signer: owner1 ,
    transaction,
  });

version:
"@aptos-labs/ts-sdk": "^1.26.0",
"@aptos-labs/wallet-adapter-react": "^3.4.3",

When i get the account data from wallet, i can't use in aptos.transaction.sign.
The console showed the error msg:
TypeError: a.signTransactionWithAuthenticator is not a function

if i use:
let owner2 = Account.generate();
const bobSenderAuthenticator = aptos.transaction.sign({
    signer: owner2 ,
    transaction,
  });

results:
no error

Therefore, how can i change the account type to Ed25519Account ?

**Comment by 0xmaayan** [comment]
Hi @MartinYeung5 ! 

To sign a transaction, you must provide the signer/wallet private key so it can be used to sign the transaction with.

When you try to do it with the info the wallet adapter provides, you wont succeed as the wallet adapter NEVER exposes the wallet private key (for obvious security reasons), so
```
const { account } = useWallet();
```
only returns the public wallet info such as the account address, public key, etc. Because of that, when you provide that argument to the SDK `sign` method, you are getting an error (not sure if you use TypeScript but you should get a type error when trying to pass a different type to that function).

On the other hand, when you generate a new account with `let owner2 = Account.generate();` that account type holds the private key and when you are passing it to the SDK `sign` method it succeeds. 

Not sure what the exact flow you are trying to achieve, 
- If you only want to sign and submit transaction in a one go, you could use `signAndSubmitTransaction` method from the `useWallet()` provider.
- If you want to just sign a transaction using the SDK `aptos.transaction.sign` method, then you would need to know the signer private key and build an `Account` out of it 
```
const privateKey = new Ed25519PrivateKey("0xc533...fa5");
const account = Account.fromPrivateKey({ privateKey });
const bobSenderAuthenticator = aptos.transaction.sign({
  signer: account,
  transaction,
});
```

**Reply by MartinYeung5** [reply]
thanks for your reply with detailed explanation,
but if i want to let user to do the sign action, how can i do ?
Actually, i can't get the user privatekey
